packages/atlas/src/api/hooks/nfts.ts

```diff
@@ -7,11 +11,82 @@ import {
 } from '@/api/queries'
 import { createLookup } from '@/utils/data'
 
-export const useNft = (id: string) => {
+export type NftStatus =
+  | {
+      status: 'auction'
+      type: 'open-auction' | 'english-auction'
+      startingPrice: number
+      buyNowPrice?: number
+      topBid?: number
+      topBidder?: BasicMembershipFieldsFragment
+      title?: string | null
+      duration?: number | null
+      views?: number
+      auctionPlannedEndBlock?: number
+      startsAtDate?: Date
+    }
+  | {
+      title?: string | null
+      duration?: number | null
+      status: 'idle'
+      lastPrice?: number
+      lastTransactionDate?: Date
+      views?: number
+    }
+  | {
+      title?: string | null
+      duration?: number | null
+      status: 'buy-now'
+      buyNowPrice: number
+      views?: number
+    }
+export type Nft = AllNftFieldsFragment | null
+export type UseNftData = Omit<QueryResult, 'data'> & { nft?: Nft; nftStatus: NftStatus }
+
+export const useNft = (id: string): UseNftData => {
   const { data, ...rest } = useGetNftQuery({ variables: { id }, skip: !id })
+  const nft = data?.ownedNftByUniqueInput
+
+  const commonProperties = {
+    title: nft?.video?.title,
+    duration: nft?.video?.duration,
+    views: nft?.video?.views,
+  }
+
+  const getNftProperties = (): NftStatus => {
+    switch (nft?.transactionalStatus.__typename) {
+      case 'TransactionalStatusAuction': {
+        return {
+          ...commonProperties,
+          status: 'auction',
+          type:
+            nft.transactionalStatus.auction?.auctionType.__typename === 'AuctionTypeOpen'
+              ? 'open-auction'
+              : 'english-auction',
+          startingPrice: Number(nft.transactionalStatus.auction?.startingPrice) ?? 0,
+          buyNowPrice: Number(nft.transactionalStatus.auction?.buyNowPrice) ?? undefined,
```

Comment:
If `buyNowPrice` is missing, this will produce `NaN`. You could fix it by changing to `||` instead of `??`